n_helpers.c

```diff
@@ -84,6 +84,45 @@ NOTE_C_STATIC void setTime(JTIME seconds);
 NOTE_C_STATIC bool timerExpiredSecs(uint32_t *timer, uint32_t periodSecs);
 NOTE_C_STATIC int ytodays(int year);
 
+static const char BINARY_EOP = '\n';
+
+const char *NoteBinaryDecode(const uint8_t *inBuf, uint32_t inLen,
+                             uint8_t *outBuf, uint32_t *outLen)
+{
+    uint32_t decLen = cobsDecode((uint8_t *)inBuf, inLen, BINARY_EOP, outBuf);
+
+    if (outLen != NULL) {
```

Comment:
In `NoteBinaryTransmit`, we decode the user's input buffer on errors. In those cases, we have no use for the output length, so I deliberately made this function permit `outLen == NULL`.

Reply:
Yes, if we can ascertain the lower bound on the required length for the decoded data, that's preferable to just trying and hoping it doesn't crash. 

Is there a better name for `cobsGuaranteedFit()`? To me, it's not immediately obvious this is talking about decoding space required, and still wasn't all that clear after I read the doc comments.

Reply:
I agree, but that is Ray's naming convention and it is taken directly from the Notecard firmware. My goal is to keep these files identical.

Changing the name is a reasonable request, but it should be done outside of this PR.